topiary/tests/samples/expected/nickel.ncl

```diff
@@ -1372,37 +1369,35 @@
       # Returns the sub-record of `l` containing only those fields which are not
       # present in `r`. If `l` has a sealed polymorphic tail then it will be
       # preserved.
-      let field_diff =
-        fun left right =>
-          array.fold_left
-            (
-              fun acc field =>
-                if %has_field% field right then
-                  acc
-                else
-                  %record_insert% field acc (left."%{field}")
-            )
-            (%record_empty_with_tail% left)
-            (%fields% left)
+      let field_diff = fun left right =>
+        array.fold_left
+          (
+            fun acc field =>
+              if %has_field% field right then
+                acc
+              else
+                %record_insert% field acc (left."%{field}")
+          )
+          (%record_empty_with_tail% left)
+          (%fields% left)
       in
       let contracts_not_in_value = field_diff field_contracts value in
       let missing_fields = %fields% contracts_not_in_value in
       if %length% missing_fields == 0 then
         let tail_fields = field_diff value field_contracts in
-        let fields_with_contracts =
-          array.fold_left
-            (
-              fun acc field =>
-                if %has_field% field field_contracts then
-                  let contract = field_contracts."%{field}" in
-                  let label = %go_field% field label in
-                  let val = value."%{field}" in
-                  %record_insert% field acc (%assume% contract label val)
-                else
-                  acc
-            )
-            {}
-            (%fields% value)
+        let fields_with_contracts = array.fold_left
```

Comment:
I'm wondering why a newline isn't inserted here? isn't the child an application?

Reply:
I found this quite curious, too... I've just dug into it and I _think_ I know the reason. The query to determine the exceptions is not strict enough and will match any sibling rather than just the first-sibling. In this case, there are three exceptions -- two parentheticals and one record -- that the query matches and thus, incorrectly, deduces the RHS to be an exception.

Let me fix that :sweat_smile:

Reply:
...Yep, that was definitely the problem. Now fixed: https://github.com/tweag/topiary/pull/423/commits/f894cf2a3ae88963dea75018812f00b857ed438a#diff-6986669f19c1caad9af03e15df1e512864cc90f72071dee180866ea9dc1210e0